examples/example_sdl_opengl3/main.cpp

```diff
@@ -9,6 +9,16 @@
 #include <stdio.h>
 #include <SDL.h>
 
+// GL includes
+#if defined(IMGUI_IMPL_OPENGL_ES2)
+#include <GLES2/gl2.h>
+#elif defined(IMGUI_IMPL_OPENGL_ES3)
+#if (defined(__APPLE__) && (TARGET_OS_IOS || TARGET_OS_TV))
```

Comment:
Where do `TARGET_OS_IOS` and `TARGET_OS_TV` come from? I do not see them in Makefile. Does platform define them?

Reply:
They're in `<TargetConditionals.h>`. Include is missing.

Reply:
TARGET_OS_XX macros are defined by Apple headers, specifically `TargetConditionals.h`.
What's not absolutely clear is if that file is included even indirectly from main.cpp, maybe including it explicitely would be good (like imgui_impl_opengl3.cpp does)

Reply:
But if we can establish that e.g. SDL.h always include it I'm happy relying on that knowledge.

Reply:
Seems that compiler / pre-processor define those. Some insight here: http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system#OSXiOSandDarwin 

It would be nice to detect 'platforms' like RPi like this but as it happens, to compiler RPi CPU looks just like any other ARM based CPU. While ARM CPU can be distinguished from x86, it is too wide.

Reply:
If `__APPLE__` is defined `"TargetConditionals.h"` is safe to include and I think it should be. This is the place where all `TARGET_XXX` macros are defined.
If SDL.h include it it would be for very same reason it should be included here. Those are macros and they are leaking into user code, relying on that does sound like a good idea.

This type of target detection isn't unique to Apple platforms. Windows SDK provide `<winapifamily.h>` serving very same purpose.

Reply:
what about `grep -o BCM2708 /proc/cpuinfo` for RPI detection? (or BCM2709 for the rpi 2 & 3). However this would not work when cross compiling. Perhaps another approach is to[ check for /opt/vc/include/bcm_host.h](https://raspberrypi.stackexchange.com/questions/754/how-can-i-detect-that-im-compiling-for-raspberry-pi)